server/state/index.js

```diff
@@ -23,21 +23,30 @@ const initSmartContracts = () => {
   smartContracts = List()
 }
 
+const isDuplicateSmartContract = smartContract => {
+  if (smartContracts.find(x => x.smartContract === smartContract)) {
```

Comment:
Any time you find yourself doing:
```
if() return true
else return false
```
you can just retunr what is in the if statement:

```js
const isDuplicate = smartContract => smartContracts.find(x => x.smartContract === smartContract
```

Reply:
Also, let's think about naming here, think for future devs:

when you map through `smartContracts`, then `x` is really a single `smartContract`. Personally I think it makes most sense to rename the `smartContract` field to `address`

that way when you're looping through smartContracts, you can do `smartContract.address` and it reads fluidly